File: LeetCode/cpp/2233_Maximum_Product_After_K_Increments.cpp
class Solution {
public:
	int maximumProduct(vector<int>& nums, int k) {
		int answer = 1;

		priority_queue<int, vector<int>, greater<>> lo;

		for (int n : nums) {
			lo.push(n);
		}

		while (k > 0) {
			int n = lo.top();
			lo.pop();
			n++;

			lo.push(n);
			k--;
		}

		while (!lo.empty()) {
			long long n = lo.top() % 1000000007;
			answer = (answer * n) % 1000000007;
			lo.pop();
		}

		return answer;
	}
};

File: LeetCode/java/2233_Maximum_Product_After_K_Increments.java
import java.util.PriorityQueue;

class Solution {
    public int maximumProduct(int[] nums, int k) {
        int answer = 1;

        PriorityQueue<Integer> lo = new PriorityQueue<>();

        for (int n : nums) {
            lo.add(n);
        }

        while (k > 0) {
            int n = lo.poll();
            n++;

            lo.add(n);
            k--;
        }

        while (!lo.isEmpty()) {
            long n = lo.poll() % 1000000007;
            answer = (int)((answer * n) % 1000000007);
        }

        return answer;
    }
}